### DISCUSSION TITLE
Drawing circles in the background

### DISCUSSION THREAD

**Discussion by royfrancis** [typst/typst#2708] [Q&A] [answered]
I am trying to create this background on an A4 paper.

![bg](https://github.com/typst/typst/assets/5083500/047ac54e-b0dc-4c15-b961-8470f7cc610f)

I suppose I could use [circle](https://typst.app/docs/reference/visualize/circle/), but how do I position it exactly? And how do I place it in the background? ie; everything else I place on the document must be above it and the circles must not interact with the flow of the document (text, other images etc).

**Comment by andreasKroepelin** [comment, marked as answer]
You can use [the `background` parameter of `page`](https://typst.app/docs/reference/layout/page/#parameters-background) in combination with [`place`](https://typst.app/docs/reference/layout/place/).

**Reply by royfrancis** [reply]
Thanks! How would I combine multiple [contents](https://typst.app/docs/reference/foundations/content/) together? Like a container? div() for example.

```
  set page(
    margin: (left: 2.5cm, right: 2.5cm, top: 2.5cm, bottom: 2cm),
    background: (
        place(top + left, circle(radius: 12cm, fill: rgb("#95b540"), stroke: none)),
        place(top + right, circle(radius: 5cm, fill: rgb("#95b540"), stroke: none))
    )
  )
```

**Reply by andreasKroepelin** [reply]
Use `{ }` instead of `( )` for `background`:
```
...
background: {
  place(...)
  place(...)
}
...
```

**Reply by andreasKroepelin** [reply]
Note that you can also deduplicate your drawing code by using `set`:

```
background: {
  set circle(fill: rgb("#123456"), stroke: none)
  place(top, circle(radius: 3cm))
  place(bottom, circle(radius: 4cm))
}
```
